### ISSUE TITLE
pre-traind model

### ISSUE THREAD

**Issue by beibeirory** [yuxumin/PoinTr#22]
In you experiments, you have compared your model with 'PF-Net' and 'GRNet'. Could you provide the pre-trained models of 'PF-Net' and 'GRNet' on PCN and KITTI datasets? We don't have enough resources to trian the model. Thanks a lot.

**Comment by yuxumin** [comment]
Hi, we haven't compared PoinTr with PF-Net on PCN.  As for the performance for PF-Net on KITTI, we directly use the results from previous papers. (We haven't train PF-Net on PCN or KITTI). And the author of GRNet provides the pretrained weight on PCN and KITTI [here](https://github.com/hzxie/GRNet#pretrained-models).

**Comment by beibeirory** [comment]
Thanks. I have another question. Could you tell me the average CD L1 loss  of the finetuned model on PCNCars, or provide the finetuned model?

**Comment by yuxumin** [comment]
Do you mean the PoinTr model finetuned on PCNCars? It is named PoinTr on KITTI. ( Since one can't train the completion models on KITTI.)

**Comment by beibeirory** [comment]
Sorry, I get it.

**Comment by beibeirory** [comment]
I tested the pretrained model on KITTI dataset. But I got "MMD is 0.000539". Do you have any idea about that?

**Comment by yuxumin** [comment]
Hi, the performance is reported as 0.526(e-3) in our paper . Last time I test this pretrained model getting MMD of 0.504(e-3). 
I think It is normal to get a slightly different performance for each independent run.

**Comment by beibeirory** [comment]
Thank you very much.

**Comment by beibeirory** [comment]
I'm sorry to disturb you again. Could you release the code of PFNet? I see you have compared with PFNet on ShapeNet-55 and ShapeNet-34.

**Comment by yuxumin** [comment]
I'm sorry that we have no plans to implement PFNet into our framework recently since it is different with other methods (In PFNet,  two networks are defined and optimized). We report the results of PFNet on ShapeNet-34/55 based on their official code. What you only need to do is  to change the dataset with our ShapeNet-34/55 and train the network according to their official code.

**Comment by beibeirory** [comment]
When I finetune you pretrained PCN model on KITTI, the losses  largely fluctuate up and down (the minimum loss is more than 9.0). Could you tell any training tricks?

**Comment by yuxumin** [comment]
We use random-drop of a relative large ratio to help the model adapt to the situation on KITTI. I check the final CD for my KITTI model, it seems around 8.0.
No more tricks, sorry.

**Comment by beibeirory** [comment]
Sorry, I can not fully understand "We use random-drop of a relative large ratio to help the model adapt to the situation on KITTI". Could you show me the code?

**Comment by yuxumin** [comment]
see [here](https://github.com/yuxumin/PoinTr/blob/master/tools/runner.py#L87)

**Comment by beibeirory** [comment]
I see it. Thanks a lot. I will check the code.
